src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -39,6 +39,28 @@ def _get_versions(self, location):
         create_version = next(x for x in versions if not x.startswith(prefix))
         return create_version, upgrade_version
 
+    def generate_ssh_keys(self):
+        # If the `--ssh-key-value` option is not specified, the validator will try to read the ssh-key from the "~/.ssh" directory,
+        # and if no key exists, it will call the method provided by azure-cli.core to generate one under the "~/.ssh" directory.
+        # In order to avoid misuse of personal ssh-key during testing and the race condition that is prone to occur when key creation
+        # is handled by azure-cli when performing test cases concurrently, we provide this function as a workround.
+
+        # In the scenario of runner and AKS check-in pipeline, a temporary ssh-key will be generated in advance under the "/opt/.ssh"
+        # directory when setting up the environment. Each test case will read the ssh-key from a pre-generated file during execution,
+        # so there will be no race conditions caused by concurrent reading and writing/creating of the same file.
+        pre_generated_ssh_key_path = "/opt/.ssh/id_rsa.pub"
```

Comment:
is /opt/.ssh a good place?
why not just using some place like the "test" data or so.

Reply:
Because this workaround is applicable to both azure-cli and azure-cli-extensions, it is not appropriate to set the default ssh-key storage directory as the test subdirectory of any of the aforementioned *repos*. Now I have change the default ssh-key storage directory to `/tmp/azaks/.ssh`.